### PR TITLE
Move to a PyPI JSON API endpoint containing package versions

### LINKED ISSUE #41: When packaging older versions, some metadata (e.g. license) is obtained from the latest one
We've noticed with @befeleme that when a custom version is specified, some metadata only relevant to the latest release is obtained from PyPI API.

More specifically, whenever pyp2conf.py accesses `self.package_data["info"]`, the information is relevant to the latest release and not to the one the packager tries to package.

Examples:

https://github.com/befeleme/pyp2spec/blob/71d6dae7d85cacf0c47d5a49aebb0357d6996573/pyp2spec/pyp2conf.py#L171-L173

https://github.com/befeleme/pyp2spec/blob/71d6dae7d85cacf0c47d5a49aebb0357d6996573/pyp2spec/pyp2conf.py#L134

### PR DESCRIPTION
**Pull request by befeleme**
This is more complex to unpack, better to look at particular commits.
- version is newly handled from within the PypiPackage object, not as an external factor. It's an inherent attribute of a package, either set by an user, or obtained from the metadata (the latest one)
- with that in place, move to the new API to handle package requests. If we rely on PyPI to retrieve a version, one request is made to the previous API endpoint to retrieve the latest version string, and then the metadata belonging to that version are requested from the new API
- too many `session`s flying around. I think requests to PyPI can be made from one session saved as an attribute of the class and updated the code accordingly. 
Fixes #41

### CODE REVIEW

File: pyp2spec/pyp2conf.py [hunk 1]
```diff
@@ -43,26 +45,46 @@ def pypi_name(self):
 
         return self.normalize(self.package_data["info"]["name"])
 
+    @property
+    def version(self):
+        """Set either to the custom package version provided by the user
+        or to the latest one from package metadata.
+        """
+
+        return self._version or self._get_version_from_package_metadata()
```
Comment (hroncok) [inline pyp2spec/pyp2conf.py:54]: This property seems to query PyPI every time when read (unless `self._version` is set).

That seems wasteful and might create a race condition when  new version is uploaded to PyPI mid-run.

What if the property did something like this:

```suggestion
        self._version = self._version or self._get_version_from_package_metadata()
        return self._version
```

Obviously, the docstring would need updating.

Comment (befeleme) [reply pyp2spec/pyp2conf.py:54]: shoot, good catch

Comment (befeleme) [reply pyp2spec/pyp2conf.py]: remedied in: b635a56819600c0e1d6ef5481154736f26aaf7f5

File: pyp2spec/pyp2conf.py [hunk 2]
```diff
@@ -27,12 +27,14 @@ class PackageNotFoundError(Pyp2specError):
 class PypiPackage:
     """Store and process the package data obtained from PyPI."""
 
-    def __init__(self, package_name, *, package_metadata=None, session=None):
+    def __init__(self, package_name, *, version=None, package_metadata=None, session=None):
         self.package_name = package_name
+        self._version = version
```
Comment (hroncok) [inline pyp2spec/pyp2conf.py:32]: Idea: Maybe the property is redundant? If the init calls this once instead, the "data flow" could be simpler.

```suggestion
        self.version = version or self._get_version_from_package_metadata()
```

**Pull request merged**